sway-core/src/ir_generation/function.rs

```diff
@@ -2970,7 +2972,7 @@ impl<'eng> FnCompiler<'eng> {
 
         // Do the actual work. This is a recursive function because we want to drill down
         // to load each primitive type in the storage field in its own storage slot.
-        self.compile_storage_read(context, ix, &field_idcs, &base_type, span_md_idx)
+        self.compile_storage_read(context, ns, ix, &field_idcs, &base_type, span_md_idx)
```

Comment:
Nitpick. Just a slight inconsistency, `compile_storage_access` accepts `fields`, `ix`, and `ns` in that order, whereas other methods go for `ns`, `ix`, `indices` (gotten from `fields`) essentially other way around (and more logical to me, to start with the namespace).